### PR TITLE
Link underline colour

### PR DESCRIPTION
**Pull request by chris-lawton**
[Link to Ticket](https://torchbox.monday.com/boards/1192293412/pulses/1472020372)

### Description of Changes Made

- Update all links to have an underline of ` rgba(255, 255, 255, 0.5)` or `#989b9b`
- Don't apply this to navigation links in the dropdowns as the new colour does not pass colour contrast - stick with the colour of the text
- Update some instances where the `link-styles()` mixin was not used

### How to Test

- Browse the site and make sure all link underlines look consistent
- Might be easier to test components on the styleguide page in the pattern library

### Screenshots

There's too many components that this has changed to provide screenshots.

### MR Checklist

- [x] Add a description of your pull request and instructions for the reviewer to verify your work.
- [x] If your pull request is for a specific ticket, link to it in the description.
- [x] Stay on point and keep it small so the merge request can be easily reviewed.
- [x] Tests and linting passes.

#### Unit tests

- [ ] Added
- [x] Not required

#### Documentation

- [ ] Updated build docs
- [ ] Updated editor guidelines (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [ ] Updated field spec (See https://intranet.torchbox.com/torchbox-com-project-docs for the private link, for Torchbox employees only)
- [x] Not required

#### Browser testing

- [ ] I have tested in the following browsers and environments (edit the list as required)
  - Latest version of Chrome on mac
  - Latest version of Firefox on mac
  - Latest version of Safari on mac
  - Safari on last two versions of iOS
  - Chrome on last two versions of Android
- [ ] Not required

#### Data protection

- [ ] Not relevant
- [ ] This adds new sources of PII and documents it and modifies Birdbath processors accordingly

#### Accessibility

- [ ] Automated WCAG 2.1 tests pass
- [ ] HTML validation passes
- [ ] Manual WCAG 2.1 tests completed
- [ ] I have tested in a screen reader
- [ ] I have tested in high-contrast mode
- [ ] Any animations removed for prefers-reduced-motion
- [ ] Not required

#### Sustainability

- [ ] Images are optimised and lazy-loading used where appropriate
- [ ] SVGs have been optimised
- [ ] Perfomance and transfer of data considered
- [ ] If JavaScript is needed alternatives have been considered
- [ ] Not required

#### Pattern library

- [ ] The pattern library component for this template displays correctly, and does not break parent templates
- [ ] The styleguide is updated if relevant
- [ ] Changes are not relevant the pattern library

### CODE REVIEW

**Comment by helenb** [comment]
@chris-lawton I think this MR in particular would be good to pull into the light mode integration branch, so that we can test the changes in light mode as well - and because it involves adding a new colour, and editing some of the same files. It's a non-urgent one, so doesn't matter if it gets a bit delayed waiting for light mode to be merged. Would you be OK to rebase against the integration branch and then do a new pull request to that branch?

File: tbx/static_src/sass/components/navigation/_sub-nav-desktop-mini.scss [hunk 1]
```diff
@@ -43,7 +43,8 @@ Here we refer to 'level 1 children' as 'level 2' and 'level 2 children' as 'leve
     &__link {
         @include link-styles(
             $color: var(--color--heading),
-            $interaction-color: var(--color--heading)
+            $interaction-color: var(--color--heading),
+            $underline-color: var(--color--heading)
```
Comment (helenb) [inline tbx/static_src/sass/components/navigation/_sub-nav-desktop-mini.scss:47]: I think the navs also needs to change to the underline colour, if we are to follow ben's request (check contrast though)

File: tbx/static_src/sass/config/_variables.scss [hunk 2]
```diff
@@ -184,6 +186,7 @@ $color--earth-streamfield-background: #e6e3df;
     --color--text-50: #{color.adjust($color--grey-70, $alpha: -0.5)};
     --color--decoration: #{$color--eclipse};
     --color--link: #{$color--eclipse};
+    --color--link-underline: #{$color--eclipse};
```
Comment (helenb) [inline tbx/static_src/sass/config/_variables.scss:189]: Maybe double check with ben what this colour should be - we might want it to be a bit lighter in comparison to the text colour.

File: tbx/static_src/sass/components/navigation/_sub-nav-desktop-mini.scss (hunk 1 above)
Comment (chris-lawton) [reply tbx/static_src/sass/components/navigation/_sub-nav-desktop-mini.scss:47]: The main nav should be updated. The new colour he suggested using doesn't pass colour contrast on the child links in the dropdown's that use a different background colour.

File: tbx/static_src/sass/config/_variables.scss (hunk 2 above)
Comment (chris-lawton) [reply tbx/static_src/sass/config/_variables.scss:189]: I've asked him on the ticket

**Review by helenb — APPROVED** [review]
Looks good!

**Pull request merged**